### PR TITLE
[OpInfo] add reference and error inputs for `multilabel_margin_loss`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #105302
* #105645
* __->__ #105523

### CODE REVIEW

File: test/test_nn.py [hunk 1]
```diff
@@ -9247,7 +9247,6 @@ def v(fn):
 
             zeros = torch.zeros_like(input).to(torch.int64)
             v(lambda: F.multilabel_soft_margin_loss(input, zeros, reduction=reduction))
-            v(lambda: F.multilabel_margin_loss(input, zeros, reduction=reduction))
```
Comment (nkaretnikov) [inline test/test_nn.py:9250]: Invalid reductions are tested in error inputs.

File: test/test_torch.py [hunk 2]
```diff
@@ -715,23 +715,6 @@ def test_scalar_check(self, device):
...
-                    self.assertEqual((), torch.nn.functional.multilabel_margin_loss(input, target, reduction='mean').shape)
-                    self.assertEqual((), torch.nn.functional.multilabel_margin_loss(input, target, reduction='sum').shape)
-                else:
-                    self.assertRaises(RuntimeError,
-                                      lambda: torch.nn.functional.multilabel_margin_loss(input, target, reduction='none'))
-                    self.assertRaises(RuntimeError,
-                                      lambda: torch.nn.functional.multilabel_margin_loss(input, target, reduction='mean'))
-                    self.assertRaises(RuntimeError,
-                                      lambda: torch.nn.functional.multilabel_margin_loss(input, target, reduction='sum'))
-
```
Comment (nkaretnikov) [inline test/test_torch.py:734]: Scalars are tested in sample/reference inputs.

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -1542,19 +1542,74 @@ def reference_inputs_like_fns(op, device, dtype, requires_grad, **kwargs):
...
+                     error_regex=(
+                         r'inconsistent target size: \[4\] for input of size: \[5, 4\]'
+                         if torch.device(device).type == 'cuda' else
+                         r'inconsistent size \[4\] for argument #2 \'target\''))
+    yield ErrorInput(SampleInput(make_input(5, 4), args=(make_input((),),), kwargs={}),
+                     error_type=RuntimeError,
+                     error_regex=(
+                         r'inconsistent target size: \[\] for input of size: \[5, 4\]'
+                         if torch.device(device).type == 'cuda' else
+                         r'inconsistent size \[\] for argument #2 \'target\''))
```
Comment (nkaretnikov) [inline torch/testing/_internal/common_methods_invocations.py:1611]: CPU/CUDA error message divergence will be fixed in a separate PR.

**Review by ezyang — APPROVED** [review]
